java/maven/test/unit/src/org/netbeans/modules/maven/runjar/RunJarPrereqCheckerTest.java

```diff
@@ -63,4 +69,103 @@ public void testWriteMapping() throws Exception {
         assertEquals(1, mapping.getActions().size());
     }
 
+    public void testMainClassManifest() throws Exception {
+        TestFileUtils.writeFile(d, "pom.xml",
+                "<project>\n" +
+                "    <modelVersion>4.0.0</modelVersion>\n" +
+                "    <groupId>testgrp</groupId>\n" +
+                "    <artifactId>testart</artifactId>\n" +
+                "    <version>1.0</version>\n" +
+                "    <build>\n" +
+                "        <plugins>\n" +
+                "            <plugin>\n" +
+                "                <groupId>org.apache.maven.plugins</groupId>\n" +
+                "                <artifactId>maven-jar-plugin</artifactId>\n" +
+                "                <configuration>\n" +
+                "                    <archive>\n" +
+                "                        <manifest>\n" +
+                "                            <mainClass>com.mycompany.Main2</mainClass>\n" +
```

Comment:
Does this also work with a property? E.g.  `<mainClass>${clazz}</mainClass>` - I assume it does, but it would be good to verify that.

Reply:
Yes, it does. A test added.